### ISSUE TITLE
I don't understand the function of "fixtheta" in the STN Module.

### ISSUE THREAD

**Issue by tanpengjie** [MediaBrain-SJTU/RegAD#9]
Dear author

Thank you very much for sharing your source code. But I don't understand the below code in the STN Module:

 tmp = np.tile(np.array([0, 0, 1]), (x.shape[0], 1, 1)).astype(np.float32)
 fixtheta1 = torch.from_numpy(np.linalg.inv(np.concatenate((theta1.detach().cpu().numpy(), tmp), axis=1))[:,:-1,:]).cuda()
 self.stn1_output = self._fixstn(x.detach(), fixtheta1)

I will appreciate it if you can help me.

**Comment by chaoqinhuang** [comment]
It is used to calculate the corresponding inverse transform matrix for each STN module. For anomaly localization, the inverse transform matrix needs to be used to re-match the regions of spatial-transformed features and the original images.

**Comment by tanpengjie** [comment]
Dear author

Thank you for your reply.
But I don't understand that "the inverse transform matrix needs to be used to re-match the regions of spatial-transformed features and the original images".  I'm extremely confused about this. Can you explain it or supply some material about it for me?  I will appreciate it, if you can help me.

**Comment by chaoqinhuang** [comment]
After STN, e.g., rotate or scaling, the spatial relationship will be changed. Thus, if the model predicts one region in the feature as anomaly, we should apply the inverse transform to find the correct anomalous region.
